Assets/ModScripts/Submodules/PolygonalMapping.cs

```diff
@@ -0,0 +1,448 @@
+            return;
+        }
+
+        if(FinalOrder[0] == "A" + y.ToString()){
+            //good press
+            Debug.LogFormat("[The Cruel Modkit #{0}] You pressed Alphabet {1}, good.", ModuleID, y);
+            Module.StartCoroutine(AnimateButtonPress(Module.Alphabet[y].transform , Vector3.down * 0.003f, 1));
+            Module.Alphabet[y].transform.Find("KeyLED").GetComponentInChildren<Renderer>().material = Module.KeyLightMats[2];
+            IsDepressed[y+6] = true;
+            FinalOrder.RemoveAt(0);
+        } else {
+            //bad press
+            Module.StartCoroutine(Module.ButtonStrike(false, y));
+            Debug.LogFormat("[The Cruel Modkit #{0}] You pressed Alphabet {1}, wrong.", ModuleID, y);
+            Module.CauseStrike();
+        }
+
+        if(FinalOrder.Count == 0){
+            Module.Solve();
+            Debug.LogFormat("[The Cruel Modkit #{0}] Module solved, good job.", ModuleID);
+        }
+
+    }
+
+    string ReturnPolygonVertexes(Vector2[] Vertexes){
+        string VertexList = "";
+        //find first label in reverse chinese order bc of how atan2 works
+        int FirstIndex = 0;
+        int FirstPosition = 261;
+        for(int i = 0; i < Vertexes.Length; i++){
+            if(Vertexes[i].x*10 + (10- Vertexes[i].y) < FirstPosition){
+                FirstPosition = (int)(Vertexes[i].x*10 + (10- Vertexes[i].y));
+                FirstIndex = i;
+            }
+        }
+
+        int NextIndex = 999;
+        int CurrentIndex = FirstIndex;
+        double MaxAtan2 = -4;
+        double Atan2StepCap = 4;
+
+        while(NextIndex != FirstIndex && VertexList.Length != 8){
+            for(int i = 0; i < Vertexes.Length; i++){
+                if(i == CurrentIndex) continue;
+                if((int)Vertexes[i].x == 777) continue;
+                double CurrentAtan2 = ReturnAtan2(Vertexes[CurrentIndex], Vertexes[i]);
+                if(CurrentAtan2 == MaxAtan2) return "6";
+
+                if(CurrentAtan2 > MaxAtan2 && CurrentAtan2 < Atan2StepCap){
+                    NextIndex = i;
+                    MaxAtan2 = CurrentAtan2;
+                }
+            }
+
+            VertexList+= NextIndex;
+            Atan2StepCap = MaxAtan2;
+            CurrentIndex = NextIndex;
+            MaxAtan2 = -4;
+        }
+        return VertexList;
+    }
+
+    bool IsInPolygon(Vector2 SymbCoord){
+        Vector2[] CoordinatesAndTest = new Vector2[7];
+        Array.Copy(Coordinates, CoordinatesAndTest, 6);
+        CoordinatesAndTest[6] = new Vector2(SymbCoord.x, SymbCoord.y);
+        if(ReturnPolygonVertexes(CoordinatesAndTest).IndexOf("6") == -1)
+            return true;
+        else
+            return false;
+
+    }
+
+    int CoordinatesFindDupe(){
+        HashSet<Vector2> DistinctCoordinates = new HashSet<Vector2>();
+        for(int i = 0; i < 6; i++)
+            if(!DistinctCoordinates.Add(Coordinates[i]))
+                return i;
+        return -1;
+    }
```

Comment:
Well, one problem with using `Coordinates.Distinct().Length == Coordinates.Length` is that I also need the index of the duplicate coord. I suppose I can marginally optimize the function with something like the following, but idk man.
`    int CoordinatesFindDupe(){
        if( Coordinates.Distinct().ToArray().Length == Coordinates.Length)
            return -1;

        HashSet<Vector2> DistinctCoordinates = new HashSet<Vector2>();
        for(int i = 0; i < 6; i++)
            if(!DistinctCoordinates.Add(Coordinates[i]))
                return i;
        return -1;
    }`

Reply:
My bad for that one, I've made this comment before seeing what it really does.

The best way to do this is instead to have another method returning a bool, and use it where it's better to check if there are duplicates rather than where that duplicate is.